### PR TITLE
Remove deprecated Laplacian API

### PR DESCRIPTION
**Pull request by ZedThree**

### CODE REVIEW

File: src/invert/laplace/invert_laplace.cxx [hunk 1]
```diff
@@ -71,45 +71,56 @@ Laplacian::Laplacian(Options *options, const CELL_LOC loc, Mesh *mesh_in)
...
+  if (maxmode < 0) {
+    maxmode = 0;
+  }
+  if (maxmode > ncz / 2) {
+    maxmode = ncz / 2;
+  }
+
+  low_mem = (*options)["low_mem"]
+                .doc("If true, reduce the amount of memory used")
+                .withDefault(false);
```
Comment (johnomotani) [inline src/invert/laplace/invert_laplace.cxx]: It looks like `low_mem` is only actually implemented now by `LaplacePDD`. If we remove that (#1828) we can remove the option too.

File: src/invert/laplace/invert_laplace.cxx [hunk 2]
```diff
@@ -71,45 +71,56 @@ Laplacian::Laplacian(Options *options, const CELL_LOC loc, Mesh *mesh_in)
...
+  all_terms =
+      (*options)["all_terms"].doc("Include first derivative terms?").withDefault(true);
 
-  if (options->isSet("flags")) {
-    if ( options->isSet("global_flags") || options->isSet("inner_boundary_flags") || options->isSet("outer_boundary_flags") ) {
-      throw BoutException("Should not use old flags as well as new global_flags/inner_boundary_flags/outer_boundary_flags");
-    }
-    int flags = (*options)["flags"]
-                    .doc("Flags to control inner and outer boundaries.")
-                    .withDefault(0);
```
Comment (johnomotani) [inline src/invert/laplace/invert_laplace.cxx]: I agree with removing this option in v5, but it might be a good idea to put a deprecation warning on use of the `laplace:flags` option in v4.3.1/v4.4. I can't think of anything better than printing a message to `output_warn` here, which isn't very noticable, but better than nothing...

File: src/invert/laplace/invert_laplace.cxx [hunk 3]
```diff
@@ -71,45 +71,56 @@ Laplacian::Laplacian(Options *options, const CELL_LOC loc, Mesh *mesh_in)
...
-    Laplacian::setFlags(flags);
-  }
-  else {
-    OPTION(options, global_flags, 0);
-    OPTION(options, inner_boundary_flags, 0);
-    OPTION(options, outer_boundary_flags, 0);
-  }
+  global_flags = (*options)["global_flags"].doc("Default flags").withDefault(0);
```
Comment (johnomotani) [inline src/invert/laplace/invert_laplace.cxx]: "Default flags" seems a bit vague, although I'm struggling to come up with something more descriptive - maybe something like "Flags to set Laplace solver options (apart from boundary conditions)"?

File: tests/integrated/test-laplace/test_laplace.cxx [hunk 4]
```diff
@@ -12,74 +12,95 @@ int main(int argc, char **argv) {
...
 
-  Field3D flag0 = invert_laplace(input, 0);
-  Field3D flag3 = invert_laplace(input, 3);
+  auto lap = std::unique_ptr<Laplacian>{Laplacian::create()};
+
+  lap->setCoefA(0.0);
+  lap->setCoefC(1.0);
+  lap->setCoefD(1.0);
+  Field3D flag0 = lap->solve(input);
+  lap->setInnerBoundaryFlags(INVERT_DC_GRAD + INVERT_AC_GRAD);
```
Comment (johnomotani) [inline tests/integrated/test-laplace/test_laplace.cxx]: It's a bit worrying that we don't test the outer-boundary-condition flags here. It'd be nice to at least set both inner- and outer-flags to the same thing. Then we'd have to update the benchmark data as well. Maybe there's a way to hack around that though... `input` and `a` are even in `(x-0.5)`, while `c` is odd, so if we duplicate each test, setting the outer boundary flags instead of the inner boundary flags then (remembering the `c` term is `1/C * Grad_perp(C)*Grad_perp(f)`, so it's sign doesn't flip even if `c`'s does) we'd have a new solution `f2(x-0.5,z)=f(-(x-0.5),z)`, where `f(x,z)` is the solution with the inner boundary condition set. So in `runtest` I think we'd just have to reverse the x-dimension of the stored solution (for the inner-boundary set case) to get the expected solution with outer boundary flags set.

File: src/invert/laplace/invert_laplace.cxx (hunk 2 above)
Comment (ZedThree) [reply src/invert/laplace/invert_laplace.cxx]: Good idea! I think there's a few options we'd like to deprecate like this too, although none spring to mind right now.

File: src/invert/laplace/invert_laplace.cxx (hunk 3 above)
Comment (ZedThree) [reply src/invert/laplace/invert_laplace.cxx]: Very true. I just took the docstring from the header.

File: tests/integrated/test-laplace/test_laplace.cxx (hunk 4 above)
Comment (ZedThree) [reply tests/integrated/test-laplace/test_laplace.cxx]: Are you ok if we hold off on fixing this now, and include it in whatever rewrite of the Laplacians we end up doing?

Comment (johnomotani) [reply tests/integrated/test-laplace/test_laplace.cxx]: yep, makes sense to me.
